### PR TITLE
add docker file, automatic builds via GitHub Actions and a short readme for manual use

### PR DESCRIPTION
**Pull request by HRA42**
As we discussed on Discord, I added a docker file, a github action for automated builds and a short read me..

You can test the image using (at least using Linux, no idea if that works for mac too?):

```
docker run -it --device /dev/snd:/dev/snd ghcr.io/hra42/pieces-voice-chat:latest
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+COPY package*.json ./
+
+# Install the Node.js dependencies
+RUN npm install
+
+# Copy the rest of the application code to the working directory
+COPY . .
+
+# Define the command to run the application
+CMD ["node", "src/index.js"]
```
Comment (mason-at-pieces) [inline Dockerfile:17]: Should we build and then run the compiled/built `index.js` instead here?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# pieces-voice-chat
+
+This is a simple voice chat CLI application that allows users to communicate with [pieces](https://pieces.app) other using their voice.
```
Comment (mason-at-pieces) [inline README.md:3]: Lets make this the following:

A simply cli that allows you to have conversations with [Pieces OS](https://docs.pieces.app/installation-getting-started/pieces-os) using just your voice.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+```
+
+2. Install the dependencies
+```bash
+npm install
+```
+
+3. Run the application
+```bash
+node src/index.js
```
Comment (mason-at-pieces) [inline README.md:20]: Also let's make this `npm run start`.

This will also require updating the `package.json` script command from `run` to `start` 👍

**Review by mason-at-pieces — CHANGES_REQUESTED** [review]
This is sweet, just a few minor updates

File: Dockerfile (hunk 1 above)
Comment (HRA42) [reply Dockerfile:17]: I don't think there is a huge difference for the size of the codebase currently, but we should keep it in mind for future reference.

**Comment by HRA42** [comment]
@mason-at-pieces I added a message for macOS and Linux..
For macOS you'll need to install pulse audio ...

1. Identify your microphone device:
   First, you need to find the device identifier for your microphone. You can do this by running the following command in Terminal:

   ```
   system_profiler SPAudioDataType
   ```

   Look for your microphone in the output and note its name.

2. Install PulseAudio:
   macOS doesn't natively support ALSA, which is commonly used for audio in Linux containers. You can use PulseAudio as a workaround. Install it using Homebrew:

   ```
   brew install pulseaudio
   ```

3. Configure PulseAudio:
   Create or edit the PulseAudio configuration file:

   ```
   nano ~/.config/pulse/default.pa
   ```

   Add these lines:

   ```
   load-module module-native-protocol-tcp auth-anonymous=1
   load-module module-esound-protocol-tcp
   load-module module-native-protocol-unix auth-anonymous=1
   ```

4. Start PulseAudio:
   Run PulseAudio in the background:

   ```
   pulseaudio --load=module-native-protocol-tcp --exit-idle-time=-1 --daemon
   ```

5. Run your Docker container:
   When running your Docker container, you'll need to pass the PulseAudio socket and set some environment variables:

   ```
   docker run -it --rm \
     -e PULSE_SERVER=docker.for.mac.localhost \
     -v ~/.config/pulse:/home/pulseaudio/.config/pulse \
     your-docker-image
   ```

6. Inside the container:
   You may need to install PulseAudio client libraries in your Docker image. For a Debian-based image, you might add:

   ```
   RUN apt-get update && apt-get install -y pulseaudio-utils
   ```

7. Test the microphone:
   Inside the container, you can test if the microphone is working using a tool like `arecord` (if installed):

   ```
   arecord -l
   ```

**Review by mason-at-pieces — APPROVED** [review]
looks good

**Pull request merged**